apiserver/keymanager/keymanager.go

```diff
@@ -113,8 +135,8 @@ func (api *KeyManagerAPI) ListKeys(arg params.ListSSHKeys) (params.StringsResult
 
 	for i, entity := range arg.Entities.Entities {
 		// NOTE: entity.Tag isn't a tag, but a username.
-		if !api.canRead(entity.Tag) {
-			results[i].Error = common.ServerError(common.ErrPerm)
+		if err := api.checkCanRead(entity.Tag); err != nil {
```

Comment:
histerical reasons - username is not even used anyway, only "admin" is ever passed in by the juu cli and it is effectively ignored IIANM. Te original implementation was all cocked up I think